components/QrCodeOverlay.tsx

```diff
@@ -7,69 +7,103 @@ import {Theme} from './ui/styleUtils';
 import {useTranslation} from 'react-i18next';
 import testIDProps from '../shared/commonUtil';
 import {SvgImage} from './ui/svg';
+// @ts-ignore
+import {generateQRData} from '@mosip/pixelpass';
+import {VerifiableCredential} from '../machines/VerifiableCredential/VCMetaMachine/vc';
+import RNSecureKeyStore, {ACCESSIBLE} from 'react-native-secure-key-store';
+import {DEFAULT_ECL, DEFAULT_QR_HEADER} from '../shared/constants';
+import {VCMetadata} from '../shared/VCMetadata';
 
 export const QrCodeOverlay: React.FC<QrCodeOverlayProps> = props => {
   const {t} = useTranslation('VcDetails');
+  const [qrString, setQrString] = useState('');
 
-  const [isQrOverlayVisible, setIsQrOverlayVisible] = useState(false);
+  async function getQRData(): Promise<string> {
+    let qrData: string;
+    try {
+      qrData = await RNSecureKeyStore.get(props.meta.id);
+    } catch {
+      let qrDataJSON = {
+        verifiableCredential: props.verifiableCredential,
+        meta: props.meta,
+      };
+      qrData = generateQRData(JSON.stringify(qrDataJSON), DEFAULT_QR_HEADER);
+      await RNSecureKeyStore.set(props.meta.id, qrData, {
+        accessible: ACCESSIBLE.ALWAYS_THIS_DEVICE_ONLY,
+      });
+    }
+    return qrData;
+  }
 
+  useEffect(() => {
+    (async () => {
+      const qrString = await getQRData();
+      setQrString(qrString);
+    })();
+  }, []);
+  const [isQrOverlayVisible, setIsQrOverlayVisible] = useState(false);
   const toggleQrOverlay = () => setIsQrOverlayVisible(!isQrOverlayVisible);
   return (
-    <React.Fragment>
-      <View testID="qrCodeView" style={Theme.QrCodeStyles.QrView}>
-        <Pressable
-          {...testIDProps('qrCodePressable')}
-          accessible={false}
-          onPress={toggleQrOverlay}>
-          <QRCode
-            {...testIDProps('qrCode')}
-            size={72}
-            value={props.qrCodeDetails}
-            backgroundColor={Theme.Colors.QRCodeBackgroundColor}
-          />
-          <View
-            testID="magnifierZoom"
-            style={[Theme.QrCodeStyles.magnifierZoom]}>
-            {SvgImage.MagnifierZoom()}
-          </View>
-        </Pressable>
-      </View>
-
-      <Overlay
-        isVisible={isQrOverlayVisible}
-        onBackdropPress={toggleQrOverlay}
-        overlayStyle={{padding: 1, borderRadius: 21}}>
-        <Column style={Theme.QrCodeStyles.expandedQrCode}>
-          <Row pY={20} style={Theme.QrCodeStyles.QrCodeHeader}>
-            <Text
-              testID="qrCodeHeader"
-              align="center"
-              style={Theme.TextStyles.header}
-              weight="bold">
-              {t('qrCodeHeader')}
-            </Text>
-            <Icon
-              {...testIDProps('qrCodeCloseIcon')}
-              name="close"
-              onPress={toggleQrOverlay}
-              color={Theme.Colors.Details}
-              size={32}
-            />
-          </Row>
-          <Centered testID="qrCodeDetails" pY={30}>
+    qrString != '' && (
```

Comment:
If qrString is empty, are we expecting to show something to user?
Or we expect qrString will never be empty then we don't need this check

Reply:
There was a discussion around this to show some error icon if the QR could not be generated.

Reply:
I think we need UI for same.